src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RaptorRoutingRequestTransitDataCreator.java

```diff
@@ -168,15 +171,21 @@ private static List<TripPatternForDate> filterActiveTripPatterns(
     // On the first search day we want to add both TripPatternsForDate objects that start that day
     // and any previous day, while on subsequent search days we only want to add the
     // TripPatternForDate objects that start on that particular day. This is to prevent duplicates.
-
-    return transitLayer
-      .getTripPatternsForDate(date)
-      .stream()
-      .filter(filter::tripPatternPredicate)
-      .filter(p -> firstDay || p.getStartOfRunningPeriod().toLocalDate().equals(date))
-      .map(p -> p.newWithFilteredTripTimes(filter::tripTimesPredicate))
-      .filter(Objects::nonNull)
-      .collect(Collectors.toList());
+    // This was previously a stream, but was unrolled for improved performance.
+    Predicate<TripTimes> tripTimesPredicate = filter::tripTimesPredicate;
+    Collection<TripPatternForDate> tripPatternsForDate = transitLayer.getTripPatternsForDate(date);
+    List<TripPatternForDate> result = new ArrayList<>(tripPatternsForDate.size());
```

Comment:
```suggestion
    final Predicate<TripTimes> tripTimesPredicate = filter::tripTimesPredicate;
    final Collection<TripPatternForDate> tripPatternsForDate = transitLayer.getTripPatternsForDate(date);
    final List<TripPatternForDate> result = new ArrayList<>(tripPatternsForDate.size());
```
Maybe it's useless but since we're talking about shaving off milliseconds, I think we can help the JIT compiler by making these final, isn't it?

Reply:
I checked, and in this case the resulting bytecode is exactly the same with or without the `final` in the local variables.